components/expression/expression-renderer.js

```diff
@@ -221,18 +311,24 @@ genotet.ExpressionRenderer.prototype.drawMatrixConditionLabels_ = function() {
     this.svgConditionLabels_.selectAll('*').remove();
     return;
   }
-  // TODO(liana): Implement the below...
-  var labels = this.svgConditionLabels_.selectAll('text').data([{
-    label: 'a vertical label'
-  }]);
+  var heatmapData = this.data.matrix;
+  var conditionLabelsData = heatmapData.conditionNames;
+  var cellWidth = this.heatmapWidth_ / conditionLabelsData.length;
+  var labels = this.svgConditionLabels_.selectAll('text').data(conditionLabelsData);
   labels.enter().append('text')
+    .attr('transform', genotet.utils.getTransform([
+      this.textHeight_ / 2 + cellWidth / 2,
+      this.conditionLabelHeight_ + this.profileHeight_ - this.labelMargin_
```

Comment:
If you are adding margin to every label, then you should translate the label group (which contains all labels).